src/openapi/streaming/connection/transport/websocket-transport.js

```diff
@@ -69,31 +69,38 @@ function destroySocket() {
     this.socket = null;
 }
 
+function restartConnection() {
+    this.reconnectTimeout = null;
+    this.reconnectCount++;
+
+    createSocket.call(this);
+
+    log.debug(LOG_AREA, 'Transport reconnected');
+}
+
 function reconnect(isImmediate) {
     if (this.reconnectCount >= DEFAULT_RECONNECT_LIMIT) {
         this.stop();
         return;
     }
 
+    if (this.reconnectTimeout && !isImmediate) {
+        log.warn(LOG_AREA, 'Reconnecting when already reconnecting');

```

Comment:
return early ?

Reply:
I was a bit scared, but yes I think we should.